compiler/rustc_expand/src/expand.rs

```diff
@@ -424,13 +424,12 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
     }
 
     /// Recursively expand all macro invocations in this AST fragment.
-    pub fn fully_expand_fragment(&mut self, input_fragment: AstFragment) -> AstFragment {
+    pub fn fully_expand_fragment(&mut self, mut fragment: AstFragment) -> AstFragment {
         let orig_expansion_data = self.cx.current_expansion.clone();
         let orig_force_mode = self.cx.force_mode;
 
         // Collect all macro invocations and replace them with placeholders.
-        let (mut fragment_with_placeholders, mut invocations) =
-            self.collect_invocations(input_fragment, &[]);
+        let mut invocations = self.collect_invocations(&mut fragment, &[]);
```

Comment:
The naming (`input_fragment`, `fragment_with_placeholders`, `expanded_fragment`) was important to me when I (re)wrote this code.
Otherwise it's not clear what is actually happening with the fragment during the mutations.

Reply:
I guess we should also leave a comment explaining the difference between these fragment phases?